packages/taiko-client/prover/proof_producer/common.go

```diff
@@ -45,7 +45,8 @@ func (res *RaikoRequestProofBodyResponseV2) Validate() error {
 	if res.Data.Status == ErrZkAnyNotDrawn.Error() {
 		return ErrZkAnyNotDrawn
 	}
-	if res.Data.Proof == nil || len(res.Data.Proof.Proof) == 0 {
+	if ProofTypeZKSP1 != res.ProofType &&
```

Comment:
Let's add a comment explaining why should check SP1 here, to let others who read this code understand.